### PR TITLE
DROOLS-4488 DMN runtime listener from kmodule

### PR DESCRIPTION
**Pull request by tarilabs**
/cc @etirelli @mariofusco @baldimir

### CODE REVIEW

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/impl/DMNRuntimeImpl.java [hunk 1]
```diff
@@ -81,6 +87,30 @@
     public DMNRuntimeImpl(InternalKnowledgeBase knowledgeBase) {
         this.knowledgeBase = knowledgeBase;
         this.eventManager = new DMNRuntimeEventManagerImpl();
+        if (knowledgeBase != null) {
+            if (knowledgeBase instanceof KnowledgeBaseImpl) {
+                KnowledgeBaseImpl knowledgeBaseImpl = (KnowledgeBaseImpl) knowledgeBase;
+                KieContainerImpl kieContainer = (KieContainerImpl) knowledgeBaseImpl.getKieContainer();
+                KieBaseModelImpl kieBaseModel = (KieBaseModelImpl) kieContainer.getKieProject().getKieBaseModel(knowledgeBase.getId());
+                for (Entry<String, String> kv : kieBaseModel.getKModule().getConfigurationProperties().entrySet()) {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/impl/DMNRuntimeImpl.java:95]: @mariofusco this part could use a code review from you, please

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/classloader/DMNRuntimeListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        kfs.write(ks.getResources().newClassPathResource("Greetings.dmn", this.getClass()));
+        kfs.writeKModuleXML("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
+                            "<kmodule xmlns=\"http://www.drools.org/xsd/kmodule\">\n" +
+                            "  <configuration>\n" +
+                            "    <property key=\"org.kie.dmn.runtime.listeners.X\" value=\"com.acme.TestListener\"/>\n" +
+                            "  </configuration>\n" +
+                            "</kmodule>");
+        kfs.writePomXML(DMNClassloaderTest.getPom(releaseId,
+                                                  ks.newReleaseId("org.kie", "kie-dmn-api", Drools.getFullVersion()),
+                                                  ks.newReleaseId("org.kie", "kie-dmn-model", Drools.getFullVersion()),
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/classloader/DMNRuntimeListenerTest.java:90]: @baldimir this part is something I could use your help with, please?
As mentioned, I am not sure why do I need to enlist the transitive dependencies.
If I leave just `kie-dmn-api`, which is the only thing really needed as the Listener in the KJAR is implementing the `DMNRuntimeListener` interface, it does not work, and I need to include all the transitives..
Thanks in advance for your review/help on this one, please?

**Review by tarilabs — COMMENTED** [review]
highlighting sensible parts for the code review

**Review by mariofusco — APPROVED** [review]

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/classloader/DMNRuntimeListenerTest.java (hunk 2 above)
Comment (tarilabs) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/classloader/DMNRuntimeListenerTest.java:90]: I have tried as you suggested @baldimir to erase ~/.m2/ org kie and drools, but without those transitives I still get:
```
[ERROR] Failures: 
[ERROR]   DMNRuntimeListenerTest.testBasicListenerFromKModule:93 [Message [id=1, level=ERROR, path=pom.xml, line=0, column=0
   text=Unresolved dependency org.kie:kie-internal:7.27.0-20190828.230556-10], Message [id=2, level=ERROR, path=pom.xml, line=0, column=0
   text=Unresolved dependency org.kie:kie-api:7.27.0-20190828.230530-10], Message [id=3, level=ERROR, path=pom.xml, line=0, column=0
   text=Unresolved dependency org.kie.soup:kie-soup-maven-support:7.27.0-20190829.030111-7]]
[ERROR]   DMNRuntimeListenerTest.testBasicListenerFromKModule:93 [Message [id=1, level=ERROR, path=pom.xml, line=0, column=0
   text=Unresolved dependency org.kie:kie-internal:7.27.0-20190828.230556-10], Message [id=2, level=ERROR, path=pom.xml, line=0, column=0
   text=Unresolved dependency org.kie:kie-api:7.27.0-20190828.230530-10], Message [id=3, level=ERROR, path=pom.xml, line=0, column=0
   text=Unresolved dependency org.kie.soup:kie-soup-maven-support:7.27.0-20190829.030111-7]]
```

Comment (baldimir) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/classloader/DMNRuntimeListenerTest.java:90]: Nevermind. Approving as it is.

**Review by baldimir — APPROVED** [review]

**Pull request merged**